### PR TITLE
add forgot password block

### PR DESCRIPTION
**Pull request by lexecon**

### CODE REVIEW

File: app/src/index.jsx [hunk 1]
```diff
@@ -1,18 +1,29 @@
...
+import { stringify, parse } from 'qs';
 
 import 'reset-css';
 import 'common/css/fonts/fonts.scss';
 import 'common/css/common.scss';
 
 import App from './components/main/app/app';
 
+
+const myHistiry = qhistory(
```
Comment (ArtsiomYeliseyenka) [inline app/src/index.jsx]: myHistory

Comment (lexecon) [reply app/src/index.jsx]: fixed

File: app/src/pages/outside/loginPage/serviceVersionsBlock/serviceVersionItem/serviceVersionItem.jsx [hunk 2]
```diff
@@ -19,51 +19,53 @@
...
+    serviceName: '',
+    serviceVersion: '',
+    serviceNewVersion: '',
+    isDeprecated: false,
+  };
+  render() {
+    const { serviceName, serviceVersion, serviceNewVersion, isDeprecated, intl } = this.props;
+    const classes = {
+      'service-version-item': true,
+      depricated: isDeprecated,
```
Comment (DiscoElevator) [inline app/src/pages/outside/loginPage/serviceVersionsBlock/serviceVersionItem/serviceVersionItem.jsx]: depr**e**cated

File: app/src/index.jsx [hunk 3]
```diff
@@ -1,18 +1,29 @@
...
+import { stringify, parse } from 'qs';
 
 import 'reset-css';
 import 'common/css/fonts/fonts.scss';
 import 'common/css/common.scss';
 
 import App from './components/main/app/app';
 
+
+const myHistory = qhistory(
```
Comment (DiscoElevator) [inline app/src/index.jsx]: I think it's better to name it `queryStringHistory` or something like that.

File: app/src/pages/outside/loginPage/forgotPasswordBlock/forgotPasswordForm/forgotPasswordForm.jsx [hunk 4]
```diff
@@ -19,64 +19,77 @@
...
 import classNames from 'classnames/bind';
-import FieldWithIcon from 'components/fields/fieldWithIcon/fieldWithIcon';
-import FieldErrorHint from 'components/fields/fieldErrorHint/fieldErrorHint';
+import { reduxForm } from 'redux-form';
+import { FieldProvider } from 'components/fields/fieldProvider';
+import { FieldWithIcon } from 'components/fields/fieldWithIcon/fieldWithIcon';
+import { FieldErrorHint } from 'components/fields/fieldErrorHint/fieldErrorHint';
 import { FormattedMessage, injectIntl, intlShape, defineMessages } from 'react-intl';
-import Input from 'components/inputs/input/input';
+import { Link } from 'react-router-dom';
```
Comment (DiscoElevator) [inline app/src/pages/outside/loginPage/forgotPasswordBlock/forgotPasswordForm/forgotPasswordForm.jsx]: Please move imports from external packages above internal imports.

File: app/src/pages/outside/loginPage/forgotPasswordBlock/forgotPasswordForm/forgotPasswordForm.jsx [hunk 5]
```diff
@@ -19,64 +19,77 @@
...
+import { Input } from 'components/inputs/input/input';
 import BigButton from 'components/buttons/bigButton/bigButton';
 import PropTypes from 'prop-types';
```
Comment (DiscoElevator) [inline app/src/pages/outside/loginPage/forgotPasswordBlock/forgotPasswordForm/forgotPasswordForm.jsx]: Please move up

File: app/src/pages/outside/loginPage/forgotPasswordBlock/forgotPasswordForm/forgotPasswordForm.jsx [hunk 6]
```diff
@@ -19,64 +19,77 @@
...
 
+import { Component } from 'react';
 import classNames from 'classnames/bind';
-import FieldWithIcon from 'components/fields/fieldWithIcon/fieldWithIcon';
-import FieldErrorHint from 'components/fields/fieldErrorHint/fieldErrorHint';
+import { reduxForm } from 'redux-form';
+import { FieldProvider } from 'components/fields/fieldProvider';
+import { FieldWithIcon } from 'components/fields/fieldWithIcon/fieldWithIcon';
+import { FieldErrorHint } from 'components/fields/fieldErrorHint/fieldErrorHint';
 import { FormattedMessage, injectIntl, intlShape, defineMessages } from 'react-intl';
```
Comment (DiscoElevator) [inline app/src/pages/outside/loginPage/forgotPasswordBlock/forgotPasswordForm/forgotPasswordForm.jsx]: Please move up

File: app/src/pages/outside/loginPage/loginBlock/loginForm/loginForm.jsx [hunk 7]
```diff
@@ -26,6 +26,7 @@ import { reduxForm } from 'redux-form';
 import { FieldWithIcon } from 'components/fields/fieldWithIcon/fieldWithIcon';
 import { FieldErrorHint } from 'components/fields/fieldErrorHint/fieldErrorHint';
 import { FormattedMessage, injectIntl, intlShape, defineMessages } from 'react-intl';
+import { Link } from 'react-router-dom';
```
Comment (DiscoElevator) [inline app/src/pages/outside/loginPage/loginBlock/loginForm/loginForm.jsx]: Please move up

**Pull request merged**